pyccl/halos/concentration.py

```diff
@@ -0,0 +1,308 @@
+from .. import ccllib as lib
+from ..background import growth_factor
+from .massdef import MassDef, mass2radius_lagrangian
+from ..power import linear_matter_power, sigmaM
+import numpy as np
+
+
+class Concentration(object):
+    """ This class enables the calculation of halo concentrations.
+
+    Args:
+        mass_def (:obj:`MassDef`): a mass definition object that fixes
+            the mass definition used by this c(M)
+            parametrization.
+    """
+    name = 'default'
+
+    def __init__(self, mass_def=None):
+        if mass_def is not None:
+            if self._check_mdef(mass_def):
+                raise ValueError("c(M) relation " + self.name +
+                                 " is not compatible with mass definition" +
+                                 " Delta = %s, " % (mass_def.Delta) +
+                                 " rho = " + mass_def.rho_type)
+            self.mdef = mass_def
+        else:
+            self._default_mdef()
+        self._setup()
+
+    def _default_mdef(self):
+        """ Assigns a default mass definition for this object if
+        none is passed at initialization.
+        """
+        self.mdef = MassDef('fof', 'matter')
+
+    def _setup(self):
```

Comment:
If you remove the default mass def, then we can simplify Concentration to be an ABC and not even have a constructor.

Reply:
or this function.

Reply:
I'll wait for your response to the above. I think it makes sense to have a base constructor to avoid repetition of check mass definition -> setup